src/AutorestSwift/Code Generator/AutoRestPlugin.swift

```diff
@@ -3,25 +3,44 @@ import NIO
 import os.log
 
 class AutoRestPlugin {
-    public func start() throws -> EventLoopFuture<AutoRestPlugin> {
-        let logger = Logger(withFileName: "autorest-swift-debug.log")
-
-        let autoRestHandler = AutoRestHandler()
-        logger.logToURL("Start NIO pipe")
+    var logger: Logger!
+    var sessionId: String?
 
-        let channel = NIOPipeBootstrap(group: MultiThreadedEventLoopGroup(numberOfThreads: 5))
+    public func start() throws -> EventLoopFuture<AutoRestPlugin> {
+        logger = Logger(withFileName: "autorest-swift-debug.log")
+        logger.logToURL("Starting NIO pipeline")
+        let handler = AutoRestHandler(withPlugin: self)
+        let channel = NIOPipeBootstrap(group: MultiThreadedEventLoopGroup(numberOfThreads: 1))
             .channelInitializer { channel in
                 channel.pipeline.addTimeoutHandlers(TimeAmount.seconds(5 * 100))
                     .flatMap {
                         channel.pipeline.addFramingHandlers(framing: .jsonpos)
                     }.flatMap {
-                        channel.pipeline.addHandler(CodableCodec<JSONRequest, JSONResponse>(), name: "AutorestServer")
+                        channel.pipeline.addHandler(CodableCodec<JSONRequest, JSONResponse>(), name: "AutoRestServer")
                     }.flatMap {
-                        channel.pipeline.addHandler(JSONRPCHandler(autoRestHandler.handle))
+                        channel.pipeline.addHandler(JSONRPCHandler(handler.handle))
                     }
             }
             .withPipes(inputDescriptor: STDIN_FILENO, outputDescriptor: STDOUT_FILENO)
 
         return channel.eventLoop.makeSucceededFuture(self)
     }
+
+    public func call(method: String, params: RPCObject) -> EventLoopFuture<RPCResult> {
+        // FIXME: Fix this Implmentation
+//        if .connected != self.state {
+//             return self.group.next().makeFailedFuture(ClientError.notReady)
+//         }
+//         guard let channel = self.channel else {
+//             return self.group.next().makeFailedFuture(ClientError.notReady)
+//         }
+//         let promise: EventLoopPromise<JSONResponse> = channel.eventLoop.makePromise()
+//         let request = JSONRequest(id: NSUUID().uuidString, method: method, params: JSONObject(params))
+//         let requestWrapper = JSONRequestWrapper(request: request, promise: promise)
+//         let future = channel.writeAndFlush(requestWrapper)
+//         future.cascadeFailure(to: promise) // if write fails
+//         return future.flatMap {
+//             promise.futureResult.map { Result($0) }
+//         }
+    }
```

Comment:
I think we need a call method on the plugin to send events, as described in:
https://github.com/apple/swift-nio-examples/blob/master/json-rpc/Sources/JsonRpc/Client.swift